subiquity/server/controllers/source.py

```diff
@@ -176,3 +179,49 @@ async def POST(self, source_id: str, search_drivers: bool = False) -> None:
 
         if changed or not self._configured:
             await self.configured()
+
+    # The identity section (and user-data) section are entirely optional
+    # on Desktop, but setting the variant doesn't happen until controllers
+    # are started. For the identity controller to properly validate
+    # this behavior, we need to do look-ahead on the source controller
+    # logic for loading the available sources.
+    # See: https://github.com/canonical/subiquity/pull/1965
+    def _install_source_is_desktop(self) -> bool:
+        # Check the autoinstall config first
+        source_config = self.app.autoinstall_config.get(self.autoinstall_key)
+        if (
+            source_config is not None
+            and (id := source_config.get("id")) is not None
+            and "server" not in id
+        ):
+            return True
+
+        # In dryrun we are on server, unless you spoof with:
+        #   source:
+        #    id: desktop
+        if self.app.opts.dry_run:
+            return False
+
+        # Check the sources available the same way the source controller does
+        # but return a bool for if the variant is desktop
+        path = "/cdrom/casper/install-sources.yaml"
```

Comment:
let's move the existing magic path and reference that common definition, please, or maybe this is obsolete given the previous comment

Reply:
Yeah obsolete but I still think it's worth moving the magic var definition somewhere global